website/blog/2019-05-13-the-new-docker-image-verdaccio-4.md

```diff
@@ -0,0 +1,60 @@
+
+### Environment Variables
+
+To avoid mistakes we have renamed all environment variables being prefixed with VERDACCIO_. This will avoid future collisions and better understanding of the origin and usage of the variable. Here the full list of the new list of variables available in the new image.
+
+[table]
+
+### Support Arbitrary User IDs
+
+The previous image runs the container with the verdaccio user and group by default, being the UID created randomly within the image. Some users were experimenting issues since some environment requires usage of arbitrary user IDs for security reasons, for such reason we have introduced the environment variable `VERDACCIO_USER_ID` that allows that. 
+
+Furthermore, other optimizations can be possible, as for instance, define a different username using `VERDACCIO_USER_NAME` and such user won’t have permissions to log in by default.
+
+### Security
+
+We have followed security recommendations removing write permissions to those locations do not need to be modified, avoiding any kind of modification with the default user provided. 
+
+For instance, the code is written to `/opt/verdaccio` the verdaccio run user cannot modify the compiled resources nor config, only the /verdaccio/storage volume. The image only assign executable permissions to the binary executable required to run verdaccio.
+
+If you are not using volumes, the `VERDACCIO_USER_NAME` will have only permissions to write in the storage folder, the source code, the configuration and plugins will be read only.
+
+To provide your own configuration file, the recommended way is using Docker volumes as follow.
+
+[example volume]
+
+
+We force a specific uid **10001** for the run user, assign the root group to the locations that need to be written by the run user; if running in a normal environment, the specific id is used and permissions are correct; if running on a randomized uid environment like openshift, the non-existent user gets assigned the root group and allowed write access to relevant locations.
+
+The entrypoint will add the user to `/etc/passwd` in case the user is running as a random uid (openshift). That way, the typical tools like whoami and so can still work. 
+
+
+## Conclusions
+
+This new image has been tested in production for months and is quite stable, thus there is no fear to give it a try. We have improved in several areas, but still there are a lot to do and for that we need you. If you are DevOps do not hesitate to give us your feedback or contribute directly in discussions and future PRs to take the Verdaccio Docker image to the next level. We count on you.
+
+## Contributions
+
+We want to thanks to **Diego Louzán**, **Dimitri Koppriwa**, **Sergio Herrera**, [Ben Tucker] (https://github.com/btucker), [Michiel De Mey](https://github.com/MichielDeMey) and me [Juan Picado](https://github.com/juanpicado) for this amazing job improving the Docker image. 
```

Comment:
Link to Ben Tucker profile is broken. Also, in my (certainly not expert) opinion, "we want to thank somebody" or "we want to give thanks to somebody" sounds better than "we want to thanks to somebody".

```suggestion
We want to thank **Diego Louzán**, **Dimitri Koppriwa**, **Sergio Herrera**, [Ben Tucker](https://github.com/btucker), [Michiel De Mey](https://github.com/MichielDeMey) and me [Juan Picado](https://github.com/juanpicado) for this amazing job improving the Docker image. 
```